all/approximation/feature_network.py

```diff
@@ -41,9 +42,13 @@ def reinforce(self):
         '''
         Backward pass of the model.
         '''
-        graphs, grads = self._dequeue()
-        graphs.backward(grads)
-        self.step()
+        if self.model.requires_grad:
```

Comment:
Erm. My impression was that requires_grad is only an attribute of tensors, but the model should in general be a nn.Module object which does not have a required_grad attribute. Which is why tests are failing.